ghcide/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -44,26 +56,53 @@ type IdeResult v = ([FileDiagnostic], Maybe v)
 -- | an IdeResult with a fingerprint
 type IdeResultNoDiagnosticsEarlyCutoff  v = (Maybe ByteString, Maybe v)
 
-ideErrorText :: NormalizedFilePath -> T.Text -> FileDiagnostic
-ideErrorText = ideErrorWithSource (Just "compiler") (Just DiagnosticSeverity_Error)
+ideErrorText :: Maybe (MsgEnvelope GhcMessage) -> NormalizedFilePath -> T.Text -> FileDiagnostic
```

Comment:
This is now a somewhat odd function, since if we have the GhcMessage we could get the text from it...

Reply:
Continued in https://github.com/haskell/haskell-language-server/pull/4311#discussion_r1636130917